ms2rescore/percolator.py

```diff
@@ -98,7 +98,7 @@ def modification_mapping(
                 self._modification_label_type = "str"
                 self._modification_mapping = value
             elif all([isinstance(label, float) for label in mod_labels]):
-                self.modification_pattern = r"\[([0-9\-\.]*)\]"
+                self.modification_pattern = r"\[((\+|\-|)[0-9\-\.]*)\]"
```

Comment:
Regex pattern is constructed in a way that you can either have - or + or nothing in front of the numerical values, group 1 will always be the numerical values with the sign (if given), although should be tested that things work without a sign as well

Reply:
works like a charm ;)

Reply:
An alternative would be:
```suggestion
                self.modification_pattern = r"\[([+-]?[0-9\-.]*)\]"
```

And if the `\-` in the second part was meant to match the sign, I guess it can be dropped.